src/renderer/component/publishForm/view.jsx

```diff
@@ -1,884 +1,574 @@
-// I'll come back to this
-/* eslint-disable */
-import React from 'react';
+// @flow
+import * as React from 'react';
 import lbry from 'lbry';
 import { isNameValid, buildURI, regexInvalidURI } from 'lbryURI';
-import { Form, FormRow } from 'component/common/form';
-import Link from 'component/link';
+import { Form, FormField, FormRow, FormFieldPrice, Submit } from 'component/common/form';
+import Button from 'component/link';
 import Modal from 'modal/modal';
 import { BusyMessage } from 'component/common';
-import ChannelSection from './internal/channelSection';
-
-class PublishForm extends React.PureComponent {
-  constructor(props) {
-    super(props);
-
-    this._requiredFields = ['name', 'bid', 'meta_title', 'tosAgree'];
-
-    this._defaultCopyrightNotice = 'All rights reserved.';
-    this._defaultPaidPrice = 0.01;
-
-    this.state = {
-      id: null,
-      uri: null,
-      rawName: '',
-      name: '',
-      bid: 10,
-      hasFile: false,
-      feeAmount: '',
-      feeCurrency: 'LBC',
-      channel: 'anonymous',
-      newChannelName: '@',
-      newChannelBid: 10,
-      meta_title: '',
-      meta_thumbnail: '',
-      meta_description: '',
-      meta_language: 'en',
-      meta_nsfw: '0',
-      licenseType: '',
-      copyrightNotice: this._defaultCopyrightNotice,
-      otherLicenseDescription: '',
-      otherLicenseUrl: '',
-      tosAgree: false,
-      prefillDone: false,
-      uploadProgress: 0.0,
-      uploaded: false,
-      errorMessage: null,
-      submitting: false,
-      creatingChannel: false,
-      modal: null,
-      isFee: false,
-      customUrl: false,
-      source: null,
-      mode: 'publish',
-    };
-  }
-
-  _updateChannelList(channel) {
-    const { fetchingChannels, fetchChannelListMine } = this.props;
+import ChannelSection from 'component/selectChannel';
+import Icon from 'component/common/icon';
+import classnames from 'classnames';
+import path from 'path';
+import type { PublishParams, UpdatePublishFormData } from 'redux/actions/publish';
+import FileSelector from 'component/common/file-selector';
+import BidHelpText from './internal/bid-help-text';
+import LicenseType from './internal/license-type';
+
+type Props = {
```

Comment:
This still seems like a giant collection of `Props`. Is it possible to subdivide this into more components to facilitate easier understanding?